api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -459,51 +455,9 @@ public void validateNewProfile(Profile profile) throws BadRequestException {
     validateProfileForCorrectness(dummyProfile, profile);
   }
 
-  // Get all the profiles. Best we can do without surgery on the entity classes is to do one query
-  // per table and join them in code.
-  public List<Profile> listAllProfiles() {
-    final Set<DbUser> usersHeavy = userService.findAllUsersWithAuthoritiesAndPageVisits();
-    final Map<Long, VerifiedInstitutionalAffiliation> userIdToAffiliationModel =
-        StreamSupport.stream(verifiedInstitutionalAffiliationDao.findAll().spliterator(), false)
-            .collect(
-                ImmutableMap.toImmutableMap(
-                    (DbVerifiedInstitutionalAffiliation a) -> a.getUser().getUserId(),
-                    verifiedInstitutionalAffiliationMapper::dbToModel));
-
-    // A user may have signed the TOS many times. Pick only the latest.
-    final Multimap<Long, DbUserTermsOfService> userIdToTosRecords =
-        Multimaps.index(userTermsOfServiceDao.findAll(), DbUserTermsOfService::getUserId);
-
-    final Map<Long, DbUserTermsOfService> userIdToMostRecentTos =
-        userIdToTosRecords.asMap().entrySet().stream()
-            .collect(
-                ImmutableMap.toImmutableMap(
-                    Entry::getKey,
-                    e ->
-                        e.getValue().stream()
-                            .max(Comparator.comparing(DbUserTermsOfService::getAgreementTime))
-                            .orElse(
-                                null))); // not reachable since  Multimaps.index() will always give
-    // a non-empty value set
-
-    // The cached free tier usage aka total cost for each user, by ID
-    final Map<Long, Double> userIdToFreeTierUsage = freeTierBillingService.getUserIdToTotalCost();
-
-    final Map<Long, Double> userIdToQuota =
-        usersHeavy.stream()
-            .collect(
-                ImmutableMap.toImmutableMap(
-                    DbUser::getUserId, freeTierBillingService::getUserFreeTierDollarLimit));
-
-    return usersHeavy.stream()
-        .map(
-            dbUser ->
-                profileMapper.toModel(
-                    dbUser,
-                    userIdToAffiliationModel.get(dbUser.getUserId()),
-                    userIdToMostRecentTos.get(dbUser.getUserId()),
-                    userIdToFreeTierUsage.get(dbUser.getUserId()),
-                    userIdToQuota.get(dbUser.getUserId())))
+  public List<AdminTableUser> listAllProfiles() {
+    return userDao.getAdminTableUsers().stream()
+        .map(dbUser -> profileMapper.adminViewToModel(dbUser))
```

Comment:
I tend to prefer method  references  where possible, i.e. `profileMapper::adminViewToModel`. additionally,  you can  declare `List<AdminTableUser> adminViewsToModels(Collection<DbUser> dbUsers);` in the mapper and use that. The code isn't as pretty, but it uses a pre-allocated list for the result.